core/src/main/java/hudson/ExtensionList.java

```diff
@@ -158,7 +158,7 @@ public void addListener(@NonNull ExtensionListListener listener) {
      *
      * Meant to simplify call inside @Extension annotated class to retrieve their own instance.
      */
-    public @NonNull <U extends T> U getInstance(@NonNull Class<U> type) throws IllegalStateException {
+    public @NonNull <U extends T> U getInstance(@NonNull Class<U> type) {
```

Comment:
[Effective Java 3rd Edition §74](https://books.google.com/books?id=BIpDDwAAQBAJ) says:

> Use the Javadoc `@throws` tag to document each exception that a method can throw, but do not use the `throws` keyword on unchecked exceptions.

This PR removes the `throws` keyword on this unchecked exception (good), but it does not add the Javadoc `@throws` tag (bad).

Reply:
The dilemma is either to keep the existing code which returns unchecked exceptions or to validate the PR which fixes this case but does not declare the exception unchecked. Another solution would be to make these corrections manually. According to Sonar's estimate, it would take around 5 hours of work to remediate the code.

Reply:
It's not a dilemma for me, as I simply won't approve this PR unless the changes I requested are made. According to my estimate, it would take around 15 minutes of work to make the requested changes.

Reply:
It was just a suggestion for improvement that you are free to reject. Would you like me to close the PR?

Reply:
It's not a net improvement if it trades one problem for another. I see no reason for this PR to be closed once the requested changes are made.